Compile errors around pasteboard when importing more UIKit.

megangardner opened the pull request:
#### 6f4b9f95a84210d37c0cb1a0692b6528b30c61de
<pre>
Compile errors around pasteboard when importing more UIKit.
<a href="https://bugs.webkit.org/show_bug.cgi?id=271424">https://bugs.webkit.org/show_bug.cgi?id=271424</a>
<a href="https://rdar.apple.com/125198774">rdar://125198774</a>

Reviewed by Tim Horton and Aditya Keerthi.

Explicitly state that m_pasteboard is an AbstractPasteboard to deal with
compile errors arrising from importing more of UIKit.

* Source/WebCore/platform/PlatformPasteboard.h:

Canonical link: <a href="https://commits.webkit.org/276581@main">https://commits.webkit.org/276581@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4daa2864fbd2bf6c4d67edab0c3a211f5106c778

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44984 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24093 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47642 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40991 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/47290 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28226 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21491 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36923 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45562 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21198 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38740 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18010 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18613 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39871 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3032 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41297 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40173 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49315 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19957 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16503 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43924 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21272 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42694 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10019 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21618 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20951 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/platform/PlatformPasteboard.h:
```diff
@@ -38,6 +38,7 @@ OBJC_CLASS NSPasteboardItem;
 
 #if PLATFORM(IOS_FAMILY)
 OBJC_CLASS UIPasteboard;
+OBJC_PROTOCOL(AbstractPasteboard);
```
Don't you need to force-conform UIPasteboard to AbstractPasteboard?

megangardner committed:
Compile errors around pasteboard when importing more UIKit.
https://bugs.webkit.org/show_bug.cgi?id=271424
rdar://125198774

Reviewed by Tim Horton and Aditya Keerthi.

Explicitly state that m_pasteboard is an AbstractPasteboard to deal with
compile errors arrising from importing more of UIKit.

* Source/WebCore/platform/PlatformPasteboard.h:

Canonical link: https://commits.webkit.org/276581@main
